core/src/test/java/test/dataprovider/issue2565/SampleTestUsingSupplier.java

```diff
@@ -0,0 +1,38 @@
+package test.dataprovider.issue2565;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Supplier;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class SampleTestClass {
```

Comment:
The class name doesn't match the file name: `SampleTestUsingSupplier`

Reply:
Dang! That was a bad commit. Missed adding the refactored class into the commit. This is now fixed.